custom/inddex/filters.py

```diff
@@ -0,0 +1,197 @@
+
+    @property
+    def options(self):
+        return [
+            ('male', _('Male')),
+            ('female', _('Female'))
+        ]
+
+
+class PregnancyFilter(BaseSingleOptionFilter):
+    slug = 'pregnant'
+    label = _('Pregnancy')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('yes', _('Yes')),
+            ('no', _('No')),
+            ('refuse_to_answer', _('Refuse to answer')),
+            ('dont_know', _('Don\'t know'))
+        ]
+
+
+class SettlementAreaFilter(BaseSingleOptionFilter):
+    slug = 'urban_rural'
+    label = _('Urban/Rural')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('urban', _('Urban')),
+            ('rural', _('Rural'))
+        ]
+
+
+class BreastFeedingFilter(BaseSingleOptionFilter):
+    slug = 'breastfeeding'
+    label = _('Breastfeeding')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        # TODO: Check value in production apllication ?? refuse_de_rpondre
+        return [
+            ('yes', _('Yes')),
+            ('no', _('No')),
+            ('refuse_de_rpondre', _('Refuse to answer'))
+        ]
+
+
+class SupplementsFilter(BaseSingleOptionFilter):
+    slug = 'supplements'
+    label = _('Supplement Use')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('yes', _('Yes')),
+            ('no', _('No'))
+        ]
+
+
+class RecallStatusFilter(BaseSingleOptionFilter):
+    slug = 'recall_status'
+    label = _('Recall Status')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('Open', _('Not Completed')),
+            ('Completed', _('Completed'))
+        ]
+
+
+class GapDescriptionFilter(BaseSingleOptionFilter):
+    slug = 'gap_description'
+    label = _('Gap description')
+    default_text = _('All')
+
+    @property
+    def gaps_descriptions(self):
+        return [
+            ('conv factor available', 'conv factor available'),
+            ('fct data available', 'fct data available'),
+            ('no conversion factor available', 'no conversion factor available'),
+            ('no fct data available', 'no fct data available'),
+            ('not applicable', 'not applicable'),
+            ('using conversion factor from base term food code', 'using conversion factor from base term food code'),
+            ('using fct data from base term food code', 'using fct data from base term food code'),
+        ]
+
+    @property
+    def options(self):
+        return [
+            x for x in self.gaps_descriptions
+        ]
+
+
+class GapTypeFilter(BaseSingleOptionFilter):
+    slug = 'gap_type'
+    label = _('Gap type')
+    default_text = _('All')
+
+    @property
+    def options(self):
+        return [
+            ('fct', 'fct'),
+            ('conversion factor', 'conversion factor'),
+        ]
+
+
+class FoodCodeFilter(BaseSingleOptionFilter):
```

Comment:
Could this be done with a dynamic choice list rather than a new SQL data source?